tests/test_config_generation.py

```diff
@@ -0,0 +1,63 @@
+from datetime import date, timedelta
+
+from utils.epinow2.constants import all_diseases, all_states, nssp_states_omit
+from utils.epinow2.functions import (
+    generate_task_configs,
+    generate_timestamp,
+    generate_uuid,
+    validate_args,
+)
+
+
+def test_default_config_set():
+    """Tests that the default set of configs is generated correctly."""
+    report_date = date.today()
+    max_reference_date = report_date - timedelta(days=1)
+    min_reference_date = report_date - timedelta(weeks=8)
+
+    default_args = {
+        "state": "all",
+        "disease": "all",
+        "report_date": report_date,
+        "reference_dates": [min_reference_date, max_reference_date],
+        "data_source": "nssp",
+        "data_path": "gold/",
+        "data_container": None,
+    }
+    validated_args = validate_args(**default_args)
+    as_of_date = generate_timestamp()
+    job_id = generate_uuid()
+    # Generate task-specific configs
+    task_configs, _ = generate_task_configs(
+        **validated_args, as_of_date=as_of_date, job_id=job_id
+    )
+    total_tasks_expected = len(list(set(all_states) - set(nssp_states_omit))) * len(
+        all_diseases
+    )
```

Comment:
```suggestion
    total_tasks_expected = len(set(all_states).difference(nssp_states_omit)) * len(all_diseases)
```

This is just me being unnecessary, because I slightly prefer `.difference()` more than calling minus on two sets.